src/main/java/ratpack/zipkin/internal/MethodCapturingRequestSpec.java

```diff
@@ -60,7 +67,8 @@ public RequestSpec headers(Action<? super MutableHeaders> action) throws Excepti
 
   @Override
   public RequestSpec method(HttpMethod method) {
-    capturedMethod = method;
+    this.capturedMethod = method;
+    span.set(impl.handler.handleSend(impl.injector, getHeaders(), this));
```

Comment:
What's the thinking behind having this class be responsible for invoking `handleSend` instead of having it in `ZipkinHttpClientImpl`? It seems odd to have this class "reach in" to invoke a method on the handler. 

This class (`MethodCapturingRequestSpec`). I kind of think of this class as being a workaround for Ratpack's API - `RequestSpec` doesn't allow you to actually get at the HTTP method that's being used, which is why this class is required. 

Regardless of which approach is taken, it still won't be as nice as having a proper callback. Unfortunately, as far as I know Ratpack doesn't have any callbacks. I spent some time trying to get interceptors added (https://github.com/ratpack/ratpack/issues/975) but keeping the PR in sync with changes upstream was getting harder and harder and I shelved it to work on other stuff.